File: src/test/java/racingcar/source/Car.java
package racingcar.source;

import racingcar.source.strategy.MoveStrategy;

public class Car {
    private final CarName name;
    private int position;
    public Car(final CarName name, int position) {
        this.name = name;
        this.position = position;

    }

    public void move(MoveStrategy moveStrategy) {
      if(moveStrategy.movable()){
          position += 1;
      }
    }


    public int getPosition() {
        return this.position;
    }
}

File: src/main/java/racingcar/domain/Car.kt
package racingcar.domain

import racingcar.strategy.move.MoveStrategy

data class Car(
    val name: Name,
    val position: Position = Position()
) {
    fun moveForward(isMovable: MoveStrategy): Car {
        return if (isMovable()) {
            copy(position = position.forward())
        } else {
            this
        }
    }
}
